sklearn_numba_dpex/common/kernels.py

```diff
@@ -295,15 +301,16 @@ def _make_partial_sum_reduction_2d_axis1_kernel(
 
     if fused_unary_func is None:
 
+        @dpex.func
         def fused_unary_func(x):
             return x
 
-    fused_unary_func = dpex.func(fused_unary_func)
+    else:
+        fused_unary_func = dpex.func(fused_unary_func())
```

Comment:
Is it normal to call `fused_unary_func`, here? Shouldn't we just pass is as an argument? If that's a workaround, can you document it (as temporarily)?
```suggestion
        fused_unary_func = dpex.func(fused_unary_func)
```

Reply:
I moved this trick from there to the driver file, it was easier to explain in a comment there.